src/content/app/tools/blast/views/blast-submission-results/BlastSubmissionResults.tsx

```diff
@@ -33,7 +45,99 @@ const BlastSubmissionResults = () => {
 };
 
 const Main = () => {
-  return <div>Here will be submission details</div>;
+  const location = useLocation();
+  const submissionId = location.pathname.split('/').slice(-1)[0];
+  const blastSubmission = useAppSelector((state) =>
+    getBlastSubmissionById(state, submissionId)
+  );
+
+  return (
+    <div className={styles.blastSubmissionResultsContainer}>
+      <Header submission={blastSubmission} isAnyJobRunning={false} />
+      <SequenceBox submission={blastSubmission} />
+    </div>
+  );
+};
+
+type SequenceBoxProps = {
+  submission: BlastSubmission;
+};
+
+const SequenceBox = (props: SequenceBoxProps) => {
+  const { submittedData, results: blastResults } = props.submission;
+
+  const resultsGroupedBySequence = submittedData.sequences.map((sequence) => {
+    const results = blastResults.filter((r) => r.sequenceId === sequence.id);
+    return {
+      sequence,
+      results
+    };
+  });
+
+  return (
+    <>
+      {resultsGroupedBySequence.map((data) => (
+        <div key={data.sequence.id} className={styles.sequenceBoxWrapper}>
+          <div className={styles.resultsSummaryRow}>
+            <div>Sequence {data.sequence.id}</div>
+            <div>Query sequence header...</div>
+            <div>
+              <span className={styles.label}>Against</span>{' '}
+              <span className={styles.boldText}>
+                {submittedData.species.length} species{' '}
+              </span>
+            </div>
+          </div>
+
+          {data.results.map((result) => {
+            const species = submittedData.species.filter(
+              (s) => s.genome_id === result.genomeId
+            );
+            return (
+              <SingleBlastJobResult
+                key={result.jobId}
+                species={species[0]}
+                jobId={result.jobId}
+              />
+            );
+          })}
+          <div className={styles.ruler}></div>
+        </div>
+      ))}
+    </>
+  );
+};
+
+type SingleBlastJobResultProps = {
+  jobId: string;
+  species: Species;
+};
+
+const SingleBlastJobResult = (props: SingleBlastJobResultProps) => {
+  const { data } = useFetchBlastSubmissionQuery(props.jobId);
+
+  if (!data) {
+    return null;
+  }
+
+  const { species: speciesInfo } = props;
+  return (
+    <div className={styles.resultsSummaryRow}>
+      <div>
+        <span className={styles.boldText}>{data.result.hits.length} </span>
+        <span className={styles.label}>
+          {' '}
+          {`${pluralise('hit', data.result.hits.length)}`}{' '}
```

Comment:
What's the purpose of this trailing white space?

Reply:
ooh thats probably created by linter? I will remove it.